packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/save-modal.tsx

```diff
@@ -0,0 +1,237 @@
+import IconRadio from './icon-radio';
+
+type TSaveModalForm = {
+    bot_name: string;
+    button_status: number;
+    google_drive_connected: boolean;
+    is_authorised: boolean;
+    is_mobile: boolean;
+    is_onscreen_keyboard_active: boolean;
+    is_save_modal_open: boolean;
+    icon: string;
+    text: string;
+    onConfirmSave: () => void;
+    onDriveConnect: () => void;
+    toggleSaveModal: () => void;
+    setCurrentFocus: () => void;
+    validateBotName: () => void;
+};
+
+const SaveModalForm = ({
+    bot_name,
+    button_status,
+    is_authorised,
+    onConfirmSave,
+    onDriveConnect,
+    validateBotName,
+    toggleSaveModal,
+    is_mobile,
+    is_onscreen_keyboard_active,
+    setCurrentFocus,
+}: TSaveModalForm) => (
+    <Formik
+        initialValues={{
+            is_local: true,
+            save_as_collection: false,
+            bot_name: bot_name === config.default_file_name ? '' : bot_name,
+        }}
+        validate={validateBotName}
+        onSubmit={onConfirmSave}
+    >
+        {({ values: { is_local, save_as_collection }, setFieldValue, touched, errors }) => {
+            const content_height = !is_mobile ? '500px' : `calc(100%)`;
+            return (
+                <ThemedScrollbars height={content_height} autohide>
+                    <Form className={classNames({ 'form--active-keyboard': is_onscreen_keyboard_active })}>
+                        <div className='modal__content'>
+                            <div className='modal__content-row'>
+                                <Field name='bot_name'>
+                                    {({ field }) => (
+                                        <Input
+                                            {...field}
+                                            className='save-type__input'
+                                            type='text'
+                                            placeholder={localize('Untitled Strategy')}
+                                            error={touched[field.name] && errors[field.name]}
+                                            label={localize('Strategy name')}
+                                            onFocus={e => setCurrentFocus(e.currentTarget.name)}
+                                            onBlur={() => setCurrentFocus(null)}
+                                        />
+                                    )}
+                                </Field>
+                            </div>
+                            <div className='modal__content-row'>
+                                <RadioGroup
+                                    className='radio-group__save-type'
+                                    name='is_local'
+                                    selected={(() => {
+                                        if (is_authorised && !is_local) return save_types.GOOGLE_DRIVE;
+                                        return save_types.LOCAL;
+                                    })()}
```

Comment:
```suggestion
                                    selected={(() => {
                                        if (is_authorised && !is_local) return save_types.GOOGLE_DRIVE;
                                        return save_types.LOCAL;
                                    })()}
```
why IIFE? Could be declared as a function

Reply:
done ✅